src/modules/uORB/uORBDeviceNode.cpp

```diff
@@ -661,12 +635,23 @@ int uORB::DeviceNode::update_queue_size(unsigned int queue_size)
 		return PX4_OK;
 	}
 
-	//queue size is limited to 255 for the single reason that we use uint8 to store it
-	if (_data || _queue_size > queue_size || queue_size > 255) {
+	if (_queue_size > queue_size || queue_size > 255) {
```

Comment:
Is it correct to return an error if we want to reduce the queue size? I guess it makes sense because when copying it over, we would potentially be removing queue data.

Reply:
Ok got it.